### PR TITLE
RAD-388: Add NINo page

### PR DESCRIPTION
**Pull request by ethanmills**

### CODE REVIEW

File: src/common/session.ts [hunk 1]
```diff
@@ -135,3 +123,58 @@ async function writeSessionFieldForSessionId(
...
+    expressionAttributeValues: ExpressionAttributeValues;
+    updateExpression: UpdateExpression;
+};
+
+/**
+ * Generate DynamoDB update query parameters from a partial session object
+ * @param sessionId Primary key of the record to update
+ * @param partialSession Field name: value pairs to update. Undefined values will be removed
+ *
+ * Given a partial session object, construct SET statements for all present values and REMOVE statements for all
```
Comment (Mjwillis) [inline src/common/session.ts:141]: So does this mean that if the session no longer contains a key/value pair it is removed from the dynamo db entry?

File: src/pages/national-insurance-number/app.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        };
+    }
+});
+
+const get = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+    const session = await getSession(event);
+    return renderAsHtmlResponse(event, 'template.njk', { session });
+};
+
+const post = (event: APIGatewayProxyEvent): GatewayResult =>
```
Comment (Mjwillis) [inline src/pages/national-insurance-number/app.ts:34]: This is nice, I assume we can use this pattern for other pages?

File: src/common/answer.ts [hunk 3]
```diff
@@ -4,9 +4,14 @@ export function includes<T extends U, U>(coll: ReadonlyArray<T>, el: U): el is T
     return coll.includes(el as T);
 }
 
-export type Answer = Partial<WhereDoYouLive & NationalInsuranceNumber & PensionProviders & TellCivilServicePension>;
+export type Answer = Partial<
+    WhereDoYouLive & NationalInsuranceNumberKnown & NationalInsuranceNumber & PensionProviders & TellCivilServicePension
+>;
+
+export type BooleanQuestion<T extends string> = { [key in T]: 'yes' | 'no' };
```
Comment (Mjwillis) [inline src/common/answer.ts:11]: Nice!

**Review by Mjwillis — APPROVED** [review]
Looks good to me

File: src/common/session.ts (hunk 1 above)
Comment (ethanmills) [reply src/common/session.ts:141]: Only if the key is explicitly included in the object and the value is `undefined`

File: src/pages/national-insurance-number/app.ts (hunk 2 above)
Comment (ethanmills) [reply src/pages/national-insurance-number/app.ts:34]: That's the idea 🙂

File: src/common/session.ts (hunk 1 above)
Comment (Mjwillis) [reply src/common/session.ts:141]: 👍

**Pull request merged**